Civi/RemoteEvent/Actions/SpawnEvent.php

```diff
@@ -0,0 +1,147 @@
+use Civi\ActionProvider\ConfigContainer;
+use Civi\ActionProvider\Parameter\ParameterBagInterface;
+use Civi\ActionProvider\Parameter\SpecificationBag;
+use Civi\ActionProvider\Parameter\Specification;
+use Civi\ActionProvider\Parameter\OptionGroupSpecification;
+use Civi\ActionProvider\Utils\CustomField;
+
+use CRM_ActionProvider_ExtensionUtil as E;
+use Dompdf\Exception;
+
+class SpawnEvent extends AbstractAction {
+
+  /**
+   * Returns the specification of the configuration options for the actual action.
+   */
+  public function getConfigurationSpecification(): SpecificationBag {
+
+    return new SpecificationBag([
+        new Specification('template_id', 'Integer', E::ts('Event Template ID'), FALSE, NULL, NULL, NULL, FALSE),
+    ]);
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public function getParameterSpecification(): SpecificationBag{
+    $specs = new SpecificationBag(array(
+      /**
+       * The parameters given to the Specification object are:
+       * @param string $name
+       * @param string $dataType
+       * @param string $title
+       * @param bool $required
+       * @param mixed $defaultValue
+       * @param string|NULL $fkEntity
+       * @param array $options
+       * @param bool $multiple
+       */
+      new Specification('template_id', 'Integer', E::ts('Event Template ID'), FALSE, NULL, NULL, NULL, FALSE),
+      new Specification('event_id', 'Integer', E::ts('Event ID'), FALSE, NULL, NULL, NULL, FALSE),
+      new OptionGroupSpecification('event_type', 'event_type', E::ts('Event Type'), FALSE),
+      new Specification('title', 'String', E::ts('Title'), TRUE, NULL, NULL, NULL, FALSE),
+      new Specification('description', 'String', E::ts('Description'), FALSE, NULL, NULL, NULL, FALSE),
+      new Specification('summary', 'String', E::ts('Summary'), FALSE, NULL, NULL, NULL, FALSE),
+      new Specification('start_date', 'Timestamp', E::ts('Start Date'), TRUE, NULL, NULL, NULL, FALSE),
+      new Specification('end_date', 'Timestamp', E::ts('End Date'), FALSE, NULL, NULL, NULL, FALSE),
+      new Specification('is_active', 'Boolean', E::ts('Is Active'), FALSE, 1, NULL, NULL, FALSE),
+      new Specification('is_public', 'Boolean', E::ts('Is Public'), FALSE, 0, NULL, NULL, FALSE),
+      new Specification('max_participants', 'Integer', E::ts('Max. Participants'), FALSE, NULL),
+      new Specification('event_full_text', 'String', E::ts('Text when event is full'), FALSE, NULL),
+      new Specification('waitlist_text', 'String', E::ts('Waitlist Text'), FALSE, NULL),
+    ));
+
+    $config = ConfigContainer::getInstance();
+    $customGroups = $config->getCustomGroupsForEntity('Event');
+    foreach ($customGroups as $customGroup) {
+      if (!empty($customGroup['is_active'])) {
+        $specs->addSpecification(CustomField::getSpecForCustomGroup($customGroup['id'], $customGroup['name'], $customGroup['title']));
+      }
+    }
+
+    ContactActionUtils::createAddressParameterSpecification($specs);
```

Comment:
As far as I can see, the fields specified here aren't used in `doAction()`.

Reply:
removed